SU2_CFD/include/numerics/ReadBFMInput.hpp

```diff
@@ -0,0 +1,193 @@
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <vector>
+#include "../variables/CVariable.hpp"
+
+class ReadBFMInput{
+private:
+
+  string version_input_file;                  // Blade geometric parameter input file version
+  unsigned short n_blade_rows{1};             // Number of blade rows in input file
+  vector<unsigned long> n_axial_points{};     // Number of axial data entries per blade row
+  vector<unsigned long> n_radial_points{};    // Number of radial data entries per blade row
+  vector<unsigned long> n_tangential_points{};// Number of tangential sections per blade row
+  vector<unsigned short> n_blades{};          // Number of blades in each blade row
+  vector<unsigned short> rotation_factor{};   // Rotation factor for each blade row
+  vector<string> variable_names{};            // Blade geometric parameter names
+
+  // vector<CVectorOfMatrix> *axial_coordinate = NULL; // Axial coordinate data entries
+  // vector<CVectorOfMatrix> *radial_coordinate = NULL;// Radial coordinate data entries
+  // vector<CVectorOfMatrix> *tangential_angle = NULL; // Tangential angle data entries
+
+  vector<vector<CVectorOfMatrix>> *Geometric_Parameters = NULL;  // Geometric blade parameters read from input file
+  vector<string> translated_names;                               // Variable name for each blade parameter
+  vector<pair<unsigned short, unsigned short>> name_translation; // Pairing name with variable index
+    
+  /*!
+   * \brief Size blade input data matrix according to header information
+   */
+  void AllocateMemory(){
+    // Allocate memory from heap
+    Geometric_Parameters = new vector<vector<CVectorOfMatrix>>;
+    
+    // Size according to number of blade rows
+    Geometric_Parameters->resize(n_blade_rows);
+    
+    // Looping through blade rows
+    for(unsigned short i_row = 0; i_row<n_blade_rows; ++i_row){
+      
+      // Size according to number of geometric parameters
+      Geometric_Parameters->at(i_row).resize(N_BFM_PARAMS);
+      
+      // Looping through parameters to size data structure according to number of spatial data entries
+      for(unsigned short i_param=0; i_param<N_BFM_PARAMS; ++i_param){
+        Geometric_Parameters->at(i_row).at(i_param).resize(n_tangential_points.at(i_row), n_radial_points.at(i_row), n_axial_points.at(i_row));
```

Comment:
The "at" interface is confusing because it is usually associated with maps.
Use operator[] and if you want a vector type that checks for bounds (in debug builds only) use `su2activevector` or `su2vector<some type>`